server/src/main/java/org/opensearch/search/pipeline/Pipeline.java

```diff
@@ -142,14 +155,20 @@ List<SearchResponseProcessor> getSearchResponseProcessors() {
         return searchResponseProcessors;
     }
 
-    SearchRequest transformRequest(SearchRequest request) throws Exception {
-        for (SearchRequestProcessor searchRequestProcessor : searchRequestProcessors) {
-            request = searchRequestProcessor.processRequest(request);
+    public SearchRequest transformRequest(SearchRequest request) throws Exception {
```

Comment:
I think you could keep this method package private, it is access from `PipelineRequest` only now